main/FatesHistoryInterfaceMod.F90

```diff
@@ -2108,6 +2110,36 @@ subroutine update_history_dyn(this,nc,nsites,sites)
                iscag = i_scls ! since imort is by definition something that only happens in newly disturbed patches, treat as such
                hio_mortality_understory_si_scag(io_si,iscag) = hio_mortality_understory_si_scag(io_si,iscag) + &
                     sites(s)%imort_rate(i_scls, i_pft)
+               !
+               ! fire mortality from the site-level diagnostic rates
+               hio_m5_si_scpf(io_si,i_scpf) = sum(sites(s)%fmort_rate(i_scls, i_pft,:))
+               hio_m5_si_scls(io_si,i_scls) = hio_m5_si_scls(io_si,i_scls) + sum(sites(s)%fmort_rate(i_scls, i_pft,:))
+               !
+               hio_crownfiremort_si_scpf(io_si,i_scpf) = sites(s)%fmort_rate_crown(i_scls, i_pft)
+               hio_cambialfiremort_si_scpf(io_si,i_scpf) = sites(s)%fmort_rate_cambial(i_scls, i_pft)
+               !
+               ! fire components of overall canopy and understory mortality
+               hio_mortality_canopy_si_scpf(io_si,i_scpf) = hio_mortality_canopy_si_scpf(io_si,i_scpf) + &
+                    sites(s)%fmort_rate(i_scls, i_pft,1)
```

Comment:
I think instead of using "1" and "2", we should be using parameters defined in EDTypesMod.F90.
Something like:  canopy_layer_index = 1

Also, for these understory mortality diagnostics.  We need to sum all understory layers, not just the second.

So, put a clause, if nclmax > canopy_layer_index,    sum(canopy_layer_index+1 : nclmax)

Reply:
Got it, and agreed. I will work through this. I did update other sections where 1 and 2 for canopy layers were hard coded, but missed this one. Thank you for catching it.